src/object/obj_class.c

```diff
@@ -482,3 +496,76 @@ obj_ec_codec_get(daos_oclass_id_t oc_id)
 
 	return NULL;
 }
+
+
+/**
+ * Encode (using ISA-L) a full stripe from the submitted scatter-gather list.
+ *
+ * oid		[IN]		The object id of the object undergoing encode.
+ * sgl		[IN]		The SGL containing the user data.
+ * sg_idx	[IN|OUT]	Index of sg_iov entry in array.
+ * sg_off	[IN|OUT]	Offset into sg_iovs' io_buf.
+ * parity	[IN|OUT]	Struct containing parity buffers.
+ * p_idx	[IN]		Index into parity p_bufs array.
+ */
+int
+obj_encode_full_stripe(daos_obj_id_t oid, daos_sg_list_t *sgl, uint32_t *sg_idx,
+		       size_t *sg_off, struct obj_ec_parity *parity, int p_idx)
```

Comment:
BTW, this func looks better to not in obj_class.c, can add another new file in src/object